File: 0134-gas-station/0134-gas-station.cs
public class Solution {
    public int CanCompleteCircuit(int[] gas, int[] cost) {
        var start = 0;
        var running_sum = 0;
        var total_sum = 0;

        for (int i = 0; i < gas.Length; i++){
            var excess_gas = gas[i] - cost[i];
            total_sum += excess_gas;
            running_sum += excess_gas;
            if (running_sum < 0){
                running_sum = 0;
                start = i + 1;
            }
        }
        return total_sum >= 0 ? start : -1;
    }
}

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        excess_gas = [g - c for g, c in zip(gas, cost)]
        
        start = running_sum = total_sum = 0
        
        for index in range(len(excess_gas)):
            total_sum += excess_gas[index]
            running_sum += excess_gas[index]
            if running_sum < 0:
                running_sum = 0
                start = index + 1
        
        return start if total_sum >= 0 else -1